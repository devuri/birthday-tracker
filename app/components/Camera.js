'use strict';
import React, { Component } from 'react';
import { Text, View, TouchableOpacity, Image, Platform } from 'react-native';
import styles from '../styles/main';
import ImagePicker from 'react-native-image-picker';

const Camera = (props) => {

  const selectPhotoTapped = () => {
    const options = {
      quality: 1.0,
      maxWidth: 125,
      maxHeight: 125,
      storageOptions: {
        skipBackup: true
      }
    };

    ImagePicker.showImagePicker(options, (response) => {
      console.log('Response = ', response);

      if (response.didCancel) {
        console.log('User cancelled photo picker');
      }
      else if (response.error) {
        console.log('ImagePicker Error: ', response.error);
      }
      else {
        const source = {uri: 'data:image/jpeg;base64,' + response.data, isStatic: true};
        props.setAvatar(source);
      }
    });
  };

  return (
    <View>
      <TouchableOpacity onPress={selectPhotoTapped.bind(this)}>
        <View>
        { props.avatar === null ? <Image style={styles.avatar} source={require('../img/avatar-placeholder.png')} /> :
          <Image style={styles.avatar} source={props.avatar} />
        }
        </View>
      </TouchableOpacity>
    </View>
  );
}

export default Camera;
