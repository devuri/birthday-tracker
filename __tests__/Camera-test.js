import 'react-native';
import React from 'react';
import Camera from '../app/components/Camera.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Camera />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
