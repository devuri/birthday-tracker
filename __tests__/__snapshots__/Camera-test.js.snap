exports[`test renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View>
      <Image
        source={undefined}
        style={
          Object {
            "borderRadius": 62,
            "height": 125,
            "marginBottom": 10,
            "width": 125,
          }
        } />
    </View>
  </View>
</View>
`;
